src/test/java/com/fasterxml/jackson/dataformat/xml/ser/TestSerialization.java

```diff
@@ -31,6 +31,22 @@ static class AttrAndElem
         public int attr = 42;
     }
 
+    static class Floats
```

Comment:
Good. One addition suggestion: also verify wrappers `Float` / `Double` just because handling by Jackson varies in some places (due to there being different underlying class being introspected). Should behave the same of course, but just confirming.